backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowIssueRequestHandler.java

```diff
@@ -184,23 +190,31 @@ public static class ShowIssueQuery {
     private final String tokenName;
     @Nullable
     private final String tokenValue;
+    @Nullable
+    private final String organizationKey;
 
     public ShowIssueQuery(String serverUrl, String projectKey, String issueKey, String branch,
-      @Nullable String pullRequest, @Nullable String tokenName, @Nullable String tokenValue) {
+      @Nullable String pullRequest, @Nullable String tokenName, @Nullable String tokenValue, @Nullable String organizationKey) {
       this.serverUrl = serverUrl;
       this.projectKey = projectKey;
       this.issueKey = issueKey;
       this.branch = branch;
       this.pullRequest = pullRequest;
       this.tokenName = tokenName;
       this.tokenValue = tokenValue;
+      this.organizationKey = organizationKey;
     }
 
     public boolean isValid() {
       return isNotBlank(serverUrl) && isNotBlank(projectKey) && isNotBlank(issueKey) && isNotBlank(branch)
+        && (!isSonarCloud(serverUrl) || isNotBlank(organizationKey))
         && isPullRequestParamValid() && isTokenValid();
     }
 
+    private static boolean isSonarCloud(String serverUrl) {
+      return PRODUCTION_URI.toString().equals(serverUrl);
```

Comment:
Shouldn't we respect the `SonarCloudActiveEnvironment` here? SC's URL can be overridden by clients to be able to test in different environments, and I think we should use that here